### DISCUSSION TITLE
ERROR  [uncaughtException] Invalid argument(s): Uri /node_modules/bootstrap/scss/_variables.scss must have scheme 'file:'.

### DISCUSSION THREAD

**Discussion by mrFANRA** [twbs/bootstrap#41001] [Q&A] [answered]
Hi! Have any idea why its happen?

```
"nuxt": "^3.10.3",
"sass": "1.35.1"
"bootstrap": "^5.3.3",
```

scss:
```
@import "bootstrap/scss/functions";
@import "bootstrap/scss/variables";
```

exception:
```
ERROR  [uncaughtException] Invalid argument(s): Uri /node_modules/bootstrap/scss/_variables.scss must have scheme 'file:'.                                                               19:02:55

  at Object.wrapException (node_modules\sass\sass.dart.js:1231:17)
  at WindowsStyle.pathFromUri$1 (node_modules\sass\sass.dart.js:33230:17)
  at StaticClosure.fromUri (node_modules\sass\sass.dart.js:16163:37)
  at Object.NullableExtension_andThen0 (node_modules\sass\sass.dart.js:18504:40)
  at Object._wrapException (node_modules\sass\sass.dart.js:13047:14)
  at _render_closure1.call$2 (node_modules\sass\sass.dart.js:81775:21)
  at _RootZone.runBinary$3$3 (node_modules\sass\sass.dart.js:27547:18)
  at _FutureListener.handleError$1 (node_modules\sass\sass.dart.js:26096:19)
  at _Future__propagateToListeners_handleError.call$0 (node_modules\sass\sass.dart.js:26394:49)
  at Object._Future__propagateToListeners (node_modules\sass\sass.dart.js:4541:77)
```

**Comment by julien-deramond** [comment]
I'm not entirely sure, but it looks like there might be an incompatibility between the Sass version you're using and how Nuxt 3 handles module imports. Here are a few things I'd suggest trying:
1. Update Sass version: Try upgrading your Sass dependency to a more recent version, such as `1.78.0` (current version used by Bootstrap)
2. Adjust the import paths: If upgrading Sass isn’t possible, try modifying the import paths. Instead of `@import "bootstrap/scss/functions";`, use something like `@import "@/node_modules/bootstrap/scss/functions"`; to specify the full path more explicitly.
3. Share a reproducible environment: If neither of these steps work, could you try creating a reproducible environment in a tool like StackBlitz? This can help identify the root cause more accurately and allow for targeted troubleshooting.

Hope these steps could help to narrow down the issue!

**Comment by mrFANRA** [comment]
Thanks for response!

When Im upgrade Sass to 1.78 got deprecation warnings like this:
```
Deprecation Warning: Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

    ┌──> node_modules\bootstrap\scss\_reboot.scss
503 │     font-weight: $legend-font-weight;
    │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ declaration
    ╵
    ┌──> node_modules\bootstrap\scss\vendor\_rfs.scss
136 │ ┌     @media (#{$rfs-mq-property-width}: #{$rfs-mq-value}) {
137 │ │       @content;
138 │ │     }
    │ └─── nested rule
    ╵
    node_modules\bootstrap\scss\_reboot.scss 503:3  @import
    assets\plugins\bootstrap.scss 2:9               @import
    assets\app.scss 6:9                             root stylesheet
 ```

File error happen on last line _variables.scss (sass 1.35.1). Here is:
```
@import "variables-dark"; // TODO: can be removed safely in v6, only here to avoid breaking changes in v5.3    
```
When Im comment this line, error not happen.

**Reply by julien-deramond** [reply, marked as answer]
> When Im upgrade Sass to 1.78 got deprecation warnings like this:

My bad, I gave you the Sass version of our `main` branch, not released yet. That should work with a Sass version between `1.71.0` and `1.78.0`.

**Reply by mrFANRA** [reply]
Thanks! Now is working,
